src/VisualStudio/CSharp/Impl/Snippets/SnippetCommandHandler.cs

```diff
@@ -114,5 +116,26 @@ protected override bool IsSnippetExpansionContext(Document document, int startPo
             return !syntaxTree.IsEntirelyWithinStringOrCharLiteral(startPosition, cancellationToken) &&
                 !syntaxTree.IsEntirelyWithinComment(startPosition, cancellationToken);
         }
+
+        public CommandState GetCommandState(AutomaticLineEnderCommandArgs args, Func<CommandState> nextCommandHandler)
+            => nextCommandHandler();
+
+        public void ExecuteCommand(
+            AutomaticLineEnderCommandArgs args,
+            Action nextCommandHandler,
+            CommandExecutionContext executionContext)
+        {
+            AssertIsForeground();
+            if (!AreSnippetsEnabled(args))
+            {
+                nextCommandHandler();
+                return;
+            }
+
+            if (!TryHandleTypedSnippet(args.TextView, args.SubjectBuffer))
+            {
+                nextCommandHandler();
+            }
```

Comment:
whatever this code is doing, is it handled by an integration test or unit tests which pulls in all the right handlers?